drivers/kw2xrf/kw2xrf.c

```diff
@@ -1064,30 +1064,26 @@ void kw2xrf_isr_event(gnrc_netdev_t *netdev, uint32_t event_type)
 int _assemble_tx_buf(kw2xrf_t *dev, gnrc_pktsnip_t *pkt)
 {
     gnrc_netif_hdr_t *hdr;
-    int index = 0;
+    int index = 4;
 
     if (dev == NULL) {
-        gnrc_pktbuf_release(pkt);
-        return -ENODEV;
+        return 0;
     }
 
     /* get netif header check address length */
     hdr = (gnrc_netif_hdr_t *)pkt->data;
 
-    /* FCF, set up data frame, request for ack, panid_compression */
-    /* TODO: Currently we don´t request for Ack in this device.
-     * since this is a soft_mac device this has to be
-     * handled in a upcoming CSMA-MAC layer.
-     */
-    /*if (dev->option & KW2XRF_OPT_AUTOACK) {
-        dev->buf[1] = 0x61;
-    }
-    else {
-        dev->buf[1] = 0x51;
-    }*/
-    dev->buf[1] = 0x51;
+    /* we are building a data frame here */
+    dev->buf[1] = IEEE802154_FCF_TYPE_DATA;
```

Comment:
[0] contains the size of packet
https://github.com/RIOT-OS/RIOT/pull/4264/files#diff-cae64f326f164c9b37208e6a9f3f9c8dR1211

The driver is currently being rebuilt. I will make it more understandable.